### ISSUE TITLE
Generated static www.data.gov

### ISSUE THREAD

**Issue by jbrown-xentity** [GSA/data.gov#2873]
## User Story

As a data.gov administrator, I want a low-compliance burden static site so that deployment and management cost is lowered for the development team.

## Acceptance Criteria
[ACs should be clearly demoable/verifiable whenever possible. Try specifying them using [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development#Behavioral_specifications).]
- [ ] WHEN I merge a new Markdown file in the posts/ directory onto main \
  THEN I see a new post on the preview branch in Federalist.

## Background
This comes from a similar [research ticket](https://github.com/GSA/datagov-deploy/issues/1465).
While not included in this ticket, we will need to handle #3488  before we can cut over to this new site.

Styling the jekyll site will be the bulk of the work; this will be a manual process. The whole site will share some common header/footer element. Other than root level pages, all contents can be grouped into different categories and each category will have some different UI elements that need custom layout. There are 20-30 categories, depending on what tool is used to do the counting, so there are 20-30 layouts needed to be coded in Jekyll. We have crawled the Wordpress site and will use the that list as the source for all content.

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
As the result of this story will be a static site, the security profile for www.data.gov will be greatly improved. Since the resulting site will be a Federalist site, the content will be in a public GitHub repository. We only need to ensure that any content which was not public in the DB is omitted during the content migration process, and that permissions on the repository are properly limited to just the appropriate folks.

## Sketch

1. Get familiar with the Wordpress layouts that are in https://github.com/GSA/data.gov/tree/master/roots-nextdatagov/templates (from which the static pages were captured)
1. Launch a local Jekyll site, start with a simple root level page, get UI elements exported and base layout coded into Jekyll.
1. Prioritize the missing layouts, decide which are bases for others
1. Pick some simple pages for each category and code its layout (static capture, no refactoring)
1. Refactor layouts to use templates at leisure
1. Connected the site to NetlifyCMS.

**Comment by mogul** [comment]
Ryan Wold (of the neighboring Touchpoints team) has volunteered to take a stab at this for us! I'll meet with him tomorrow and bring him up to date.

**Comment by hkdctol** [comment]
There are a lot of data.gov URLs that we would not need to keep going forward. See spreadsheet at https://docs.google.com/spreadsheets/d/1XzP6bbIa7A1Jf0mEkDdEhQfsfYfE0YG0EHOtqPQJ6HM/edit?usp=sharing

**Comment by mogul** [comment]
> There are a lot of data.gov URLs that we would not need to keep going forward. See spreadsheet at https://docs.google.com/spreadsheets/d/1XzP6bbIa7A1Jf0mEkDdEhQfsfYfE0YG0EHOtqPQJ6HM/edit?usp=sharing

We made a separate issue for this:
- #3497

**Comment by adborden** [comment]
The sketch here is fine, but I wouldn't put too much emphasis on the existing WordPress templates. It will take time to learn how those templates are structured and how to map content pages to which layout. This is effort and learning that you will end up throwing away. Ultimately, you'll want the outcome of the templates to look more like the USWDS templates, because you'll be using USWS CSS and not WordPress CSS. **Do not try to mix multiple styling frameworks on the same page.**

My gut is that the UI component patterns and styles on www.data.gov today are not that complicated. I think the straightforward sketch would be to start with USWDS templates and then add additional styling/components to make them match www.data.gov. If you run into a complex UI component, that's when looking at the WordPress templates/styles might be useful.

Because you can have multiple layout "hierarchies" within a single Jekyll site, this makes it easy to maintain legacy layouts completely separate from new USWDS layouts. The CSS/JS/assets that each layout pulls in can be completely separate too which allows you to have a multiple "web stacks" independent from each other using their own frameworks but living in the same repo and served from the same website.

**Comment by adborden** [comment]
Based on the idea from https://github.com/gsa/datagov-deploy/issues/3551, I wrote a quick [python script that crawls the WP API](https://github.com/gsa/datagov-wp-migration) and outputs markdown files for posts and pages. It's not 100% correct, but you could use these markdown to replace _some_ of the content in https://github.com/gsa/datagov-website. I would cp the markdown files over to datagov-website, run a jekyll build, and then prune any .html files that cause jekyll to complain about URL conflicts.

**Comment by adborden** [comment]
There are a variety of database dumps and wordpress exports on Drive https://drive.google.com/drive/folders/1C2f1gxAS6D37ImUBSa1sxM_gewOg7bHh

**Comment by hkdctol** [comment]
Closing it is being finished on #3891
